File: exercise/scala/day15/src/main/scala/santaChristmasList/operations/Business.scala
package santaChristmasList.operations

class Business(factory: Factory, inventory: Inventory, wishList: WishList) {
  def loadGiftsInSleigh(children: Child*): Sleigh = {
    val sleigh = new Sleigh()
    for (child <- children) {
      val gift = wishList.identifyGift(child)
      if (gift.isDefined) {
        val manufacturedGift = factory.findManufacturedGift(gift.get)
        if (manufacturedGift.isDefined) {
          val finalGift = inventory.pickUpGift(manufacturedGift.get.barCode)
          if (finalGift.isDefined) {
            sleigh.put(child, s"Gift: ${finalGift.get.name} has been loaded!")
          }
        }
      }
    }
    sleigh
  }
}

File: exercise/kotlin/day15/src/main/kotlin/santaChristmasList/operations/Business.kt
package santaChristmasList.operations

class Business(
    private val factory: Factory,
    private val inventory: Inventory,
    private val wishList: WishList
) {
    fun loadGiftsInSleigh(vararg children: Child): Sleigh {
        val sleigh = Sleigh()
        for (child in children) {
            val gift = wishList.identifyGift(child)
            if (gift != null) {
                val manufacturedGift = factory.findManufacturedGift(gift)
                if (manufacturedGift != null) {
                    val finalGift = inventory.pickUpGift(manufacturedGift.barCode)
                    if (finalGift != null) {
                        sleigh[child] = "Gift: ${finalGift.name} has been loaded!"
                    }
                }
            }
        }
        return sleigh
    }
}